Validation/MtdValidation/plugins/EtlLocalRecoValidation.cc

```diff
@@ -175,6 +176,27 @@ void EtlLocalRecoValidation::analyze(const edm::Event& iEvent, const edm::EventS
   using namespace edm;
   using namespace std;
   using namespace geant_units::operators;
+  using namespace mtd;
+
+  // Struct to  identify the pixel
+  struct ETLPixelId {
+    const uint32_t detid_;
+    const uint8_t row_;
+    const uint8_t col_;
+    ETLPixelId() : detid_(0), row_(0), col_(0) {}
+    ETLPixelId(const ETLDetId& id, uint8_t row, uint8_t col) : detid_(id.rawId()), row_(row), col_(col) {}
+    bool operator==(const ETLPixelId& other) const {
+      return detid_ == other.detid_ && row_ == other.row_ && col_ == other.col_;
+    }
+  };
+
+  struct PixelKey_hash {
+    size_t operator()(const ETLPixelId& key) const {
+      return std::hash<uint32_t>{}(key.detid_) ^ std::hash<uint8_t>{}(key.row_) ^ std::hash<uint8_t>{}(key.col_);
+    }
+  };
+
+  MTDGeomUtil geomUtil;
```

Comment:
the use of MTDGeomUtil is possible if geometry and topology are properly initialised with the set methods:

https://github.com/cms-sw/cmssw/blob/master/Geometry/MTDGeometryBuilder/interface/MTDGeomUtil.h#L28

For ETL at least the geometry needs to be set

https://github.com/cms-sw/cmssw/blob/master/Geometry/MTDGeometryBuilder/src/MTDGeomUtil.cc#L132

otherwise the code will happily crash. The comment is valid for all classes affected by this PR.